tests/integration_tests/test_xgboost.py

```diff
@@ -1,48 +1,73 @@
+from collections import OrderedDict
+
 import numpy as np
 import pytest
 import xgboost as xgb
 
 import optuna
+from optuna.integration.xgboost import use_callback_cls
 from optuna.integration.xgboost import XGBoostPruningCallback
 from optuna.testing.integration import create_running_trial
 from optuna.testing.integration import DeterministicPruner
 
 
 def test_xgboost_pruning_callback_call() -> None:
+    if use_callback_cls is False:
+        env = xgb.core.CallbackEnv(
+            model="test",
+            cvfolds=1,
+            begin_iteration=0,
+            end_iteration=1,
+            rank=1,
+            iteration=1,
+            evaluation_result_list=[["validation-error", 1.0]],
+        )
 
-    env = xgb.core.CallbackEnv(
-        model="test",
-        cvfolds=1,
-        begin_iteration=0,
-        end_iteration=1,
-        rank=1,
-        iteration=1,
-        evaluation_result_list=[["validation-error", 1.0]],
-    )
-
-    # The pruner is deactivated.
-    study = optuna.create_study(pruner=DeterministicPruner(False))
-    trial = create_running_trial(study, 1.0)
-    pruning_callback = XGBoostPruningCallback(trial, "validation-error")
-    pruning_callback(env)
-
-    # The pruner is activated.
-    study = optuna.create_study(pruner=DeterministicPruner(True))
-    trial = create_running_trial(study, 1.0)
-    pruning_callback = XGBoostPruningCallback(trial, "validation-error")
-    with pytest.raises(optuna.TrialPruned):
+        # The pruner is deactivated.
+        study = optuna.create_study(pruner=DeterministicPruner(False))
+        trial = create_running_trial(study, 1.0)
+        pruning_callback = XGBoostPruningCallback(trial, "validation-error")
```

Comment:
Hmm, I think we don't need this block because I guess the CI will install xgboost>=1.3 hereafter and this block will not be tested.

Reply:
I'm not sure if optuna wants to support both versions of xgboost.  If the police is supports only the latest one then I will remove all old tests.

Reply:
At least I want to support both versions, but I don't think it's necessary to have the tests for both implementations (and in our current CI system, it's really hard). So again, I'd say removing the test for the old version is okay.

Reply:
@crcrpar Thanks for the explanation.  Removed.